### ISSUE TITLE
OSRM creates instructions with invalid road names

### ISSUE THREAD

**Issue by mk112014** [Project-OSRM/osrm-backend#1732]
On Windows, OSRM has no problem with small data, but with large data it creates instructions with invalid road names.
Data: North America (http://download.geofabrik.de/north-america-latest.osm.pbf)
Query: viaroute?loc=37.3575027358159,-121.882495880127&loc=37.3570315637053,-121.880419850349&instructions=true
![query](https://cloud.githubusercontent.com/assets/15073906/10416522/f6b28148-7044-11e5-8865-61738ea45bb0.jpg)
Instruction:
![instruction](https://cloud.githubusercontent.com/assets/15073906/10416523/fe208fd8-7044-11e5-9b97-468f46c4b08f.jpg)

**Comment by BergWerkGIS** [comment]
Looks good on [map.project-osrm.org](http://map.project-osrm.org/?z=18&center=37.357824%2C-121.881594&loc=37.357554%2C-121.882399&loc=37.357554%2C-121.880795&hl=en&ly=&alt=&df=&srv=):

![image](https://cloud.githubusercontent.com/assets/4549888/10416567/2dbea4a6-701c-11e5-8512-1476c82586cf.png)

**Comment by BergWerkGIS** [comment]
I am able to reproduce the problem.

* osrm binaries downloaded yesterday
* north-america-latest.osm.pbf downloaded today

Output of `osrm-extract.exe` and `osrm-prepare.exe`:
https://gist.github.com/BergWerkGIS/18f4433d849b444ea29c

Wrong route (see below):
http://localhost:5000/viaroute?loc=37.3575027358159,-121.882495880127&loc=37.3570315637053,-121.880419850349&instructions=true

Correct route (see below):
http://router.project-osrm.org/viaroute?loc=37.3575027358159,-121.882495880127&loc=37.3570315637053,-121.880419850349&instructions=true


#### localhost
```json
{
  "status_message": "Found route between points",
  "status": 0,
  "found_alternative": false,
  "via_points": [
    [
      37.357498,
      -121.8825
    ],
    [
      37.357037,
      -121.880409
    ]
  ],
  "route_geometry": "uzbgfA|vbngFmIaUiZqx@tCoC~|@im@",
  "route_instructions": [
    [
      "10",
      "South Shiloh Court",
      131,
      0,
      17,
      "131m",
      "NE",
      60,
      1
    ],
    [
      "3",
      "Alkali Springs Ro",
      139,
      2,
      15,
      "139m",
      "N",
      0,
      1
    ],
    [
      "15",
      "",
      0,
      4,
      0,
      "0m",
      "N",
      0
    ]
  ],
  "route_summary": {
    "total_distance": 270,
    "end_point": "Alkali Springs Ro",
    "total_time": 28,
    "start_point": "South Shiloh Court"
  },
  "via_indices": [
    0,
    5
  ],
  "route_name": [
    "South Shiloh Court",
    "Alkali Springs Ro"
  ],
  "hint_data": {
    "checksum": 3990932950,
    "locations": [
      "o09qAOiZagBikgEAHgAAADEAAAAAAAAAAAAAAP____8AAAAAuwc6AoE4vPgAAAEB",
      "JttpACvbaQANIwAAnwAAAJQAAAAAAAAADQAAAOraYwIAAAAA7AU6AqhAvPgAAAEB"
    ]
  }
}
```


#### router.project-osrm.org
```json
{
  "hint_data": {
    "locations": [
      "PmuzANu2swAPbgQAHgAAADEAAAAAAAAAAAAAAP____8AAAAAuwc6AoE4vPgAAAEB",
      "cvayAHf2sgB6pQIAnwAAAJQAAAAAAAAADQAAAEAgeQcAAAAA7AU6AqhAvPgAAAEB"
    ],
    "checksum": 1946987922
  },
  "route_name": [
    "East Taylor Street",
    "North 20th Street"
  ],
  "via_indices": [
    0,
    5
  ],
  "found_alternative": false,
  "route_summary": {
    "end_point": "North 20th Street",
    "start_point": "East Taylor Street",
    "total_time": 28,
    "total_distance": 270
  },
  "via_points": [
    [
      37.357498,
      -121.8825
    ],
    [
      37.357037,
      -121.880409
    ]
  ],
  "route_instructions": [
    [
      "10",
      "East Taylor Street",
      131,
      0,
      17,
      "131m",
      "NE",
      60,
      1
    ],
    [
      "3",
      "North 20th Street",
      139,
      2,
      15,
      "139m",
      "N",
      0,
      1
    ],
    [
      "15",
      "",
      0,
      4,
      0,
      "0m",
      "N",
      0
    ]
  ],
  "route_geometry": "uzbgfA|vbngFmIaUiZqx@tCoC~|@im@",
  "status_message": "Found route between points",
  "status": 0
}
```

**Comment by BergWerkGIS** [comment]
Decoding

```
route_geometry: "uzbgfA|vbngFmIaUiZqx@tCoC~|@im@"
```

with [poyline](https://github.com/mapbox/polyline) returns (comma seems to be off):

```
[ [ 373.57499, -1218.82495 ],
  [ 373.57666, -1218.82142 ],
  [ 373.58103, -1218.81221 ],
  [ 373.58028, -1218.81149 ],
  [ 373.57036, -1218.80408 ] ]
```

**Comment by BergWerkGIS** [comment]
Using smaller data, [geofabrik's california-latest.osm.pbf](http://download.geofabrik.de/north-america/us/california-latest.osm.pbf) and [MapZen's san-jose_california.osm.pbf](https://s3.amazonaws.com/metro-extracts.mapzen.com/san-jose_california.osm.pbf) `start_point` and `end_point` are as expected:

<pre>
{
  "status_message": "Found route between points",
  "status": 0,
  "found_alternative": false,
  "via_points": [
    [
      37.357498,
      -121.8825
    ],
    [
      37.357037,
      -121.880409
    ]
  ],
  "route_geometry": "uzbgfA|vbngFmIaUiZqx@tCoC~|@im@",
  "hint_data": {
    "checksum": 2841167693,
    "locations": [
      "_O4AAFoHAQBUAAAAHgAAADEAAAAAAAAAAAAAAP____8AAAAAuwc6AoE4vPgAAAEB",
      "-bYAAP62AAAIAgAAnwAAAJQAAAAAAAAADQAAACblAAAAAAAA7AU6AqhAvPgAAAEB"
    ]
  },
  "route_summary": {
    "total_distance": 270,
    <b>"end_point": "North 20th Street",</b>
    "total_time": 28,
    <b>"start_point": "East Taylor Street"</b>
  },
  "via_indices": [
    0,
    5
  ],
  "route_name": [
    "",
    ""
  ]
}
</pre>

**Comment by mk112014** [comment]
Using the same data source and query, but names in your resulting instruction and mine are different.   That also appears when I builds this data on Windows systems with different RAM configure (eg. 32GB, 40GB).

**Comment by mtehver** [comment]
I encountered the issue also, on Windows. There seems to be an easy fix, though. In extraction_containers.hpp find the line
```
using STXXLStringVector = stxxl::vector<std::string>;
```
and replace it with the line
```
using STXXLStringVector = std::vector<std::string>;
```

Note that this is really a bug in the code, as stxxl containers should not be used with non POD types. Anyway, this change increases memory usage somewhat but I was able to convert whole planet with 64GB RAM/96GB pagefile in 12 hours or so.
